### ISSUE TITLE
Campaign - D.Trooper III and AT-ST insta kills Urai via reflection

### ISSUE THREAD

**Issue by AlyMar1994** [AlyMar1994/UFoCP#125]
**Quick Information:**
_OS:_ Windows 11
_FoC version:_ v1.121.786192
_UFoCP version:_ INDEV
_List any other mods used:_ N/A
_Were you playing on Steam?_ YES
_Were you in multiplayer?_ NO

**Explanation:**
I really dunno how this happens.

Did a semi-"roleplay" style campaign run. Got to the Bespin mission, attacked AT-STs and D.Trooper Phase IIIs with Urai, and Urai pretty much instantly died randomly. The Empire probably has a reflection upgrade but whether or not it should apply to Urai, or how they get it, is up for interpretation/evaluation.

**Steps to Reproduce:**
1. 

**Possible Cause:**

**Screenshots/Video:**
N/A

**Comment by cch2319** [comment]
Yeah so the reflection ability doesn't take into account lightsabers/melee, thrown projectiles like the gungan grenade and Ewoks rocks. Unless it has the Grenade identifier for the projectile it'll reflect it. But you don't want to make the melee stuff grenades because that has its own sets of behavior like gravity and arcing. Easiest fix is to make all of those units "Exceptions" listed for all the reflective upgrades on every applicable unit.
